web/vtadmin/src/api/http.ts

```diff
@@ -68,7 +66,11 @@ export const vtfetch = async (endpoint: string): Promise<HttpResponse> => {
     const json = await response.json();
     if (!('ok' in json)) throw new MalformedHttpResponseError('invalid http envelope', json);
 
-    return json as HttpResponse;
+    // Throw "not ok" responses so that react-query correctly interprets them as errors.
+    // See https://react-query.tanstack.com/guides/query-functions#handling-and-throwing-errors
+    if (!json.ok) throw new HttpResponseNotOkError(endpoint, json);
```

Comment:
"NotOkError" seems redundant to me, how about either `throw HttpResponseNotOk` or `throw HttpResponseError` ? (also if this is "just a react/ts convention" please let me know!)

Reply:
(also i know this is a type from previous PRs ....... which i probably approved without comment so .........)

Reply:
I agree it's redundant but I do find it descriptive. This error is specifically when `response.ok` (in the JSON response envelope) is false. 

There are other kinds of "HTTP response errors", like `MalformedHttpResponseError` which occurs when the response envelope is an unexpected shape. (In all cases, the `Error` suffix is there since it extends the base `Error` class.) 

I added a commit with bit of commentary on the above. If you still feel like `HttpResponseNotOkError` is redundant, I can change it.

Reply:
Cool, this makes sense to me!